commons/zenoh-core/src/macros.rs

```diff
@@ -64,6 +64,20 @@ macro_rules! zasyncread {
     };
 }
 
+// This macro performs an async read with upgrade to write option on RwLock<T>
+// For performance reasons, it first performs a try_read() and,
+// if it fails, it falls back on read().await
+#[macro_export]
+macro_rules! zasyncread_upgradable {
```

Comment:
Oh cool, I didn't realize that upgradeable reads have been added into async rwlock. :)